pisek/task_jobs/generator/manager.py

```diff
@@ -0,0 +1,220 @@
+                out_jobs[0].add_prerequisite(self._generate_inputs[seed])
+
+            jobs += inp_jobs + out_jobs
+
+        jobs += self._respects_seed_jobs(input_info, seeds, subtask)
+        return jobs
+
+    def _skip_input(self, input_info: InputInfo, seed: int, subtask: int) -> bool:
+        return not self._env.config.subtasks[subtask].new_in_subtask(
+            input_info.task_path(self._env, seed).name
+        )
+
+    def _generate_input_jobs(
+        self, input_info: InputInfo, seed: int, subtask: int, test_determinism: bool
+    ) -> list[Job]:
+        if not input_info.is_generated:
+            return []
+
+        jobs: list[Job] = [gen_inp := self._generate_input_job(input_info, seed)]
+        self._generate_inputs[seed] = gen_inp
+        input_path = input_info.task_path(self._env, seed)
+
+        if test_determinism:
+            test_det = generator_test_determinism(
+                self._env, self._env.config.in_gen, input_info, seed
+            )
+            if test_det is not None:
+                jobs.append(test_det)
+                test_det.add_prerequisite(gen_inp)
+
+        if self._env.config.in_format == DataFormat.text:
+            jobs.append(input_clean := IsClean(self._env, input_path))
+            input_clean.add_prerequisite(gen_inp)
+
+        if self._env.config.limits.input_max_size != 0:
+            jobs.append(input_small := InputSmall(self._env, input_path))
+            input_small.add_prerequisite(gen_inp)
+
+        if self._env.config.checker is not None:
+            jobs.append(
+                check_input := CheckerJob(
+                    self._env,
+                    self._env.config.checker,
+                    input_path,
+                    subtask,
+                )
+            )
+            check_input.add_prerequisite(gen_inp)
+
+        return jobs
+
+    def _generate_input_job(self, input_info: InputInfo, seed: int) -> GenerateInput:
+        gen_inp = generate_input(self._env, self._env.config.in_gen, input_info, seed)
+        self._generate_inputs[seed] = gen_inp
+        return gen_inp
+
+    def _solution_jobs(
+        self, input_info: InputInfo, seed: int, subtask: int
+    ) -> list[Job]:
+        return []
+
+    def _respects_seed_jobs(
+        self, input_info: InputInfo, seeds: list[int], subtask: int
+    ) -> list[Job]:
+        if not (input_info.is_generated and input_info.seeded):
+            return []
+
+        if not self._env.config.subtasks[subtask].new_in_subtask(
+            input_info.task_path(self._env, seeds[0]).name
+        ):
+            return []
+
+        jobs: list[Job] = []
+        if len(seeds) == 1:
+            while (seed := random.choice(SEED_RANGE)) == seeds[0]:
+                pass
+            seeds.append(seed)
+            jobs += [self._generate_input_job(input_info, seed)]
+
+        jobs.append(
+            check_seeded := GeneratorRespectsSeed(self._env, input_info, *seeds[:2])
```

Comment:
Nit: it might be useful to check all pairs, but that can wait.

Reply:
The problem with checking all pairs is that you can get a collision. (Like if input is a power of 2 lesser than 1024...)

Reply:
That can also happen if you only check two...
Maybe the correct test is if all are equal.